### PR TITLE
Fixes sheaf cohomology printing, adds check that coefficient ring is a field

### PR DESCRIPTION
**Pull request by RafaelDavidMohr**
Changes as requested in https://github.com/oscar-system/Oscar.jl/pull/3084

### CODE REVIEW

**Review by wdecker — APPROVED** [review]

File: src/Modules/ModulesGraded.jl [hunk 1]
```diff
@@ -2052,6 +2084,8 @@ end
 
 elem_type(::Type{FreeMod_dec{T}}) where {T} = FreeModElem_dec{T}
 parent_type(::Type{FreeModElem_dec{T}}) where {T} = FreeMod_dec{T}
+elem_type(::FreeMod_dec{T}) where {T} = FreeModElem_dec{T}
+parent_type(::FreeModElem_dec{T}) where {T} = FreeMod_dec{T}
```
Comment (HechtiDerLachs) [inline src/Modules/ModulesGraded.jl:2088]: Sorry, but I became curious: Where do these types come from and where are they used? I thought that the graded modules are the same as the usual modules. Did anything change there?

Comment (RafaelDavidMohr) [reply src/Modules/ModulesGraded.jl:2088]: This is beyond my scope to be honest, maybe @jankoboehm knows.

Comment (wdecker) [reply src/Modules/ModulesGraded.jl:2088]: @HechtiDerLachs Please resolve this in a direct conversation with @jankoboehm

Comment (thofma) [inline src/Modules/ModulesGraded.jl:2088]: Why are those methods necessary? https://github.com/oscar-system/Oscar.jl/pull/3068 went to great lengths to remove such methods.

CC: @lgoettgens

Comment (jankoboehm) [reply src/Modules/ModulesGraded.jl:2088]: Don't use for gradings, we plan to use this for filtrations, once they are sorted out for rings.

**Pull request merged**